src/Core/Domain/MerchandiseReturn/ValueObject/CustomizationId.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\MerchandiseReturn\Exception\MerchandiseReturnConstraintException;
+
+class CustomizationId
+{
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @param int $id
+     *
+     * @throws MerchandiseReturnConstraintException
+     */
+    public function __construct($id)
+    {
+        $this->assertIsIntegerGreaterThanZero($id);
+        $this->id = $id;
+    }
+
+    /**
+     * @return int
+     */
+    public function getValue()
+    {
+        return $this->id;
+    }
+
+    /**
+     * Validates that the value is integer and is greater than zero
+     *
+     * @param $value
+     *
+     * @throws MerchandiseReturnConstraintException
+     */
+    private function assertIsIntegerGreaterThanZero($value)
```

Comment:
Sometimes I wonder if we should have used php traits 🤔 

wdyt @zuk3975 ? I know multiple core devs dislike traits

Reply:
I can't say much about traits, but sure we could reuse this assert in some way.

Reply:
The one problem with any reuse I see is that message might be diffrent for each case. Of course you can pass part of the message or use class name for that, however I am not sure if it's worth reusing by that point.